corgidrp/l1_to_l2a.py

```diff
@@ -207,7 +208,7 @@ def detect_cosmic_rays(input_dataset, detector_params, sat_thresh=0.7,
 
 
     # Calculate the full well capacity for every frame in the dataset
-    kgain = np.array([detector_params.params['kgain'] for frame in crmasked_dataset])
```

Comment:
If kgain is None, use the value from detector_params.  And k_gain would have to be listed before detector_params in the list of arguments so that the "AUTOMATIC, OPTIONAL" works.

Reply:
By the way: is there a reason why an array of equal kgain values is made? We can divide an array by a scalar.

Reply:
agree, not really needed to have an array since we are not assuming to deal with different kgain values right now